### ISSUE TITLE
I can not create events. DB connection is read only.

### ISSUE THREAD

**Issue by tomschilling** [1and1/cosmo#35]
Hi @danielgrigore ,

I use current master branch of cosmo
Just wondering, why I can not create events.
I use a DerbyDB as Datastore. 

Always get this error message:

```
<?xml version='1.0' encoding='UTF-8'?>
<D:error xmlns:cosmo="http://osafoundation.org/cosmo/DAV" xmlns:D="DAV:">
    <cosmo:internal-server-error>org.hibernate.exception.GenericJDBCException: could not execute statement</cosmo:internal-server-error>
</D:error>
```

Looking at the stack trace, I get this SQL Error:

```
13:52:05,083 WARN [org.jboss.logging.jboss-logging] - SQL Error: 20000, SQLState: 25502
```

This means, that the connection is read only. Then I checked the belonging class `StandardContentService` where events are created and I found, that annotation  `@Transactional(readOnly = true)` 

This sets the connection into read only mode so I have no write access anymore.

From my point of view and the experience I have made with transactional behaviour in Spring context, it seems to be not right, having a read only annotation on a class that adds, copies and removes content. 

I checked with the Demo Application with MySQL and it works. 

What are your thoughts?

Cheers,

Tom

**Comment by danielgrigore** [comment]
Hi @tomschilling 

I checked the API for [Transactional](https://docs.spring.io/spring-framework/docs/current/javadoc-api/) annotation and here's what it says about the `readOnly` flag:

> This just serves as a hint for the actual transaction subsystem; it will not necessarily cause failure of write access attempts. A transaction manager which cannot interpret the read-only hint will not throw an exception when asked for a read-only transaction but rather silently ignore the hint.

The way I see it _readOnly_ is just a hint and DerbyDB should ignore it when it sees write operations are involved. (It is also true that _readOnly_ can significantly  improve performance when you don't need write access, that's why I set it).
Please let me know your thoughts.
Cheers,

**Comment by tomschilling** [comment]
Hey @danielgrigore , 

in my opinion it is a bit risky to set this read only flag. In the documentation the _not necessarily_ is italic. 

Further, I understand the second sentence 

> A transaction manager which cannot interpret...

this way - If the transaction manager can interpret the read only flag, it will cause an error. 

In fact, the JDBC connection is occupied with read only. DerbyDB shows us this case and is not ignoring the read only flag. 

I also had a glance on other DBMS. They will not ignore this flag as well. But I found that the embedded InnoDB is ignoring it. So it will definitely cause an error when using other storage engines. 

From my point of view, I would not set this read only annotation on a class that adds, copies and removes content. Just to make sure, that different transaction manager can interpret it right. 

Cheers, Tom

**Comment by danielgrigore** [comment]
Hi @tomschilling 

I investigated a little bit more the issue and tested the application with other databases and you are right.
However the problem is not related to InnoDB engine but it is related to the JDBC driver.
I used the same MariaDB instance but with `mysql-connector-java` and I experienced the problem you described above.
The same is true for PostgrSQL as far as I tested.
So I think there must be a bug in `mariadb-java-client` that does not throw an exception when writing to database in `readOnly` mode.

For this reason I modified the class `StandardContentService` so that only read operations are using the flag `readOnly` set to `true`. (The changes are visible in master now.)

Cheers,
